src/ophyd_async/core/_device.py

```diff
@@ -224,6 +224,11 @@ async def __aexit__(self, type, value, traceback):
         await self._on_exit()
 
     def __exit__(self, type_, value, traceback):
+        if in_bluesky_event_loop():
+            raise RuntimeError(
+                "Cannot use DeviceConnector inside a plan, instead use "
```

Comment:
I like this error message, it is very clear, is there any circumstance under which `in_bluesky_event_loop()` could return `True` _without_ our being inside a plan?